0.preprocess-sites/2.process-cells.py

```diff
@@ -0,0 +1,178 @@
+"""
+2.process-cells.py
+
+To view and change parameters used in this file, see `process-cells` in the config.
+
+The following script will process cell painting metadata information for each site.
+
+The metadata include:
+
+1. Cell Counts per Quality Measure
+2. Metadata Information per Cell
+
+All sites are processed independently and results are saved in site-specific folders
+(in <OUTPUT_BASEDIR>/<BATCH>/paint set in site_processing_config)
+
+1.process-spots must be run before running this script.
+"""
+
+import os
+import sys
+import pathlib
+import argparse
+import pandas as pd
+
+sys.path.append(os.path.join("..", "scripts"))
+from config_utils import process_config_file
+from cell_quality_utils import CellQuality
+from paint_utils import load_single_cell_compartment_csv, merge_single_cell_compartments
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--config_file",
+    help="configuration yaml file for preprocessing pipeline",
+    default="site_processing_config.yaml",
+)
+args = parser.parse_args()
+config_file = args.config_file
+
+config = process_config_file(config_file)
+
+# Defines the sections of the config file
+core_args = config["core"]
+cell_args = config["process-cells"]
+prefilter_args = config["prefilter"]
+spot_args = config["process-spots"]
+
+# Defines the variables set in the config file
+batch = core_args["batch"]
+batch_dir = core_args["batch_dir"]
+project_dir = core_args["project_dir"]
+quality_func = core_args["categorize_cell_quality"]
+ignore_files = core_args["ignore_files"]
+id_cols = core_args["id_cols"]
+compartments = core_args["compartments"]
+parent_cols = core_args["parent_cols"]
+
+prefilter_dir = prefilter_args["output_basedir"]
+prefilter_file = prefilter_args["prefilter_file"]
+
+foci_dir = spot_args["output_spotdir"]
+
+cell_sort_col = cell_args["sort_col"]
+output_basedir = cell_args["output_basedir"]
+merge_info = cell_args["merge_columns"]
+metadata_merge_foci_cols = cell_args["metadata_merge_columns"]["foci_cols"]
+metadata_merge_cell_cols = cell_args["metadata_merge_columns"]["cell_cols"]
+metadata_cell_quality_col = cell_args["metadata_merge_columns"]["cell_quality_col"]
+foci_site_col = cell_args["foci_site_col"]
+
+cell_quality = CellQuality(quality_func)
+cell_category_dict = cell_quality.define_cell_quality()
+empty_cell_category = len(cell_category_dict) + 1
+cell_category_dict[empty_cell_category] = "Empty"
```

Comment:
confirmed that we've added a reference in https://github.com/broadinstitute/pooled-cell-painting-profiling-recipe/issues/10#issuecomment-636196358